aioshelly/rpc_device/device.py

```diff
@@ -412,3 +419,17 @@ def connected(self) -> bool:
     def last_error(self) -> ShellyError | None:
         """Return the last error during async device init."""
         return self._last_error
+
+    @property
+    def firmware_supported(self) -> bool:
+        """Return True if device firmware version is supported."""
+        fw_ver = GEN2_MIN_FIRMWARE_DATE if self.gen == GEN2 else GEN3_MIN_FIRMWARE_DATE
+
+        match = FIRMWARE_PATTERN.search(self.firmware_version)
```

Comment:
If we are going to call this often, it might make sense to make the firmware version parser a lru_cache decorated function.

If we aren't going to call it a lot it doesn't matter

Reply:
This is called once after we initialize a device